tests/framework/db/ActiveRecordTest.php

```diff
@@ -894,6 +898,99 @@ public function testRelationWhereParams($orderTableName, $orderItemTableName)
         OrderItem::$tableName = null;
     }
 
+    public function testOutdatedRelationsAreResetForNewRecords()
+    {
+        $orderItem = new OrderItem();
+        $orderItem->order_id = 1;
+        $orderItem->item_id = 3;
+        $orderItem->quantity = 5;
+        $orderItem->subtotal = 80.0;
+        $this->assertEquals(1, $orderItem->order->id);
+        $this->assertEquals(3, $orderItem->item->id);
+
+        // Test `__set()`.
+        $orderItem->order_id = 2;
+        $orderItem->item_id = 1;
+        $this->assertEquals(2, $orderItem->order->id);
+        $this->assertEquals(1, $orderItem->item->id);
+
+        // Test `setAttribute()`.
+        $orderItem->setAttribute('order_id', 2);
+        $orderItem->setAttribute('item_id', 2);
+        $this->assertEquals(2, $orderItem->order->id);
```

Comment:
`order_id` was `2` already so this is a confusing test. Perhabs this is on purpose, in that case it would helpful to supply a third argument in the assertion to explain what you are testing (ie. 'check if order_id can be updated with it own value')